tensorflow/core/grappler/optimizers/implementation_selector.cc

```diff
@@ -262,23 +275,60 @@ absl::Status ImplementationSelector::MaybeOptimizeFunctionCall(
   }
 
   DeviceNameUtils::ParsedName parsed_name;
-  if (!DeviceNameUtils::ParseFullName(node_def->device(), &parsed_name) ||
-      !parsed_name.has_type) {
-    return errors::Internal("Could not parse device name:", node_def->device());
+  if (!node_def->device().empty()) {
+    if (!DeviceNameUtils::ParseFullName(node_def->device(), &parsed_name) ||
+        !parsed_name.has_type) {
+      return absl::InternalError(absl::StrCat(
+          "Could not parse device name: ", node_def->device()));
+    }
+    VLOG(2) << "Op " << node_def->name() << " runs on " << node_def->device()
+            << " = (" << parsed_name.type << ")";
   }
-  VLOG(2) << "Op " << node_def->name() << " runs on " << node_def->device()
-          << " = (" << parsed_name.type << ")";
+
+  auto select_device = [&](const string& function_name,
+                           const std::vector<string>& equiv_func_names) {
+    StringPiece device_type;
```

Comment:
```suggestion
```

This one is not used, results in broken builds internally (since we build with `-Werror`)